base.py

```diff
@@ -607,15 +683,22 @@ def get_table_names(self, connection, schema=None, **kw):
         Gets all table names.
         """
         schema = schema or self.default_schema_name
+        current_schema = schema
         if schema:
             cursor = connection.execute(
                 "SHOW /* sqlalchemy:get_table_names */ TABLES IN {0}".format(
                     self._denormalize_quote_join(schema)))
         else:
             cursor = connection.execute(
                 "SHOW /* sqlalchemy:get_table_names */ TABLES")
+            _, current_schema = self._current_database_schema(connection)
```

Comment:
should this call include the **kw argument to provide the info_cache?

Reply:
my mistake too. Fixing.

Reply:
Carefully looking into `_current_database_schema`, it appeared this won't work as the key is not given.